src/main/scala/dsptools/numbers/chisel_types/UIntTypeClass.scala

```diff
@@ -26,11 +26,11 @@ trait UIntRing extends Any with Ring[UInt] with hasContext {
   }
   override def minus(f: UInt, g: UInt): UInt = {
     val diff = context.overflowType match {
-      case Grow => f -& g
-      case Wrap => f -% g
+      case Grow => throw DspException("OverflowType Grow is not supported for UInt subtraction")
+      case Wrap => f.asSInt -% g.asSInt
```

Comment:
Hmm, the test case of 1 - 2 where 1 and 2 are UInt<4> returns 15 with overflow wrap, which I believe means it's being kept to4 bits.  But I also just tested line 30 as ```f -% g``` and I still got 15.